action.yml

```diff
@@ -0,0 +1,74 @@
+name: Format Markdown
+description: Automatically format changed markdown files in pull requests using Prettier.
+
+inputs:
+  github_token:
+    description: "GitHub token for pushing changes"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check event is a pull request
+      if: ${{ github.event_name != 'pull_request' }}
+      run: |
+        echo "This action only runs on pull request events. Exiting."
+        gh run cancel ${{ github.run_id }}
+        gh run watch ${{ github.run_id }}
```

Comment:
This cancels the entire workflow the action is used in, including any other steps or even jobs that are part of that workflow. Is that the intention?

If not, I don't think there's an easy way to exit early (successfully or otherwise) from an Action. Adding an `if` to each step is inelegant but would work.

Reply:
Good catch, no that isn't the intention. Yeah I'll probably opt for the `if` on each step.